dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/deprecated/tracker/importer/shared/postprocess/ProgramNotificationPostProcessor.java

```diff
@@ -63,16 +63,14 @@ public void process(
             new ProgramStageCompletionNotificationEvent(this, programStageInstance.getId()));
         applicationEventPublisher.publishEvent(
             new StageCompletionEvaluationEvent(this, programStageInstance.getUid()));
+        applicationEventPublisher.publishEvent(
```

Comment:
The fix is not so easy. Just moving the notification here will treat all the webhook events as if they have the notification trigger to COMPLETE and that is not an assumption that can be made.

Reply:
these will only trigger when event has linked web hook notifications and event status is `COMPLETED`.
Secondly web hook notifications are only suppose to work at event/enrollment completion.